src/app/configurations/allocations/calculations/calculations.states.js

```diff
@@ -21,13 +21,8 @@ angular.module('allocations')
                 return []
               })
           },
-          locationStates: function (locationService) {
-            var keys = []
-            keys.push(['2', 'NG']) // TODO: get this from auth
-            return locationService.getByLevelAndAncestor(keys)
-              .catch(function () {
-                return []
-              })
+          locationStates: function (locationService, userStateService) {
+            return userStateService.getUserSelectedState('object')
```

Comment:
Yeah, I see what you mean.

However, from perspective of someone who's reading this for the first time, I'd expect `userStateService.getUserSelectedState()` to return the state, be it a string or object. Without reading it's implementation, it's not obvious what the return value should be by passing in a parameter (true or `'object'`).

I would split it up into the following signatures:

`userStateService.getUserSelectedState`
`userStateService.getUserSelectedStateName`
`userStateService.getUserSelectedStateId`

Martin Fowler has a piece on this: [Flag argument](http://martinfowler.com/bliki/FlagArgument.html).

Sorry to derail this PR.